geocropper/utils.py

```diff
@@ -1837,3 +1837,117 @@ def move_crops_containing_locations(csv_path, source_dir, target_dir):
+    position: String
+        Set the position of the image in latitude and longitude format as a string (e.g. "-68.148781_44.77383")
+    """
+
+    image_path_list = []
+    for folder in root_dir.glob(f"*{speckle}*"):
+        if folder.is_dir() and folder.name.split("_")[0] != "stacked":
+            for crop in folder.glob(f"*{position}*"):
+                if crop.is_dir():
+                    image_path = crop / "sensordata" / "s1_cropped.tif"
+                    image_path_list.append(image_path)
+
+    return image_path_list
+
+
+def mkdir_for_stacked_image(output_dir, speckle, position):
+    """Makes a directory in the output_dir for the stacked image named by the lat_lon position and returns the 
+    path to the created directory.
+
+    Parameters
+    ----------
+    output_dir: Path
+        Path where the stacked image shall be stored
+    speckle: String
+        Set the desired speckle variant (options available in geocropper.stack_trimmed_images)
+    position: String
+        Set the position of the image in latitude and longitude format as a string (e.g. "-68.148781_44.77383")
+    """
+    
+    stacked_image_path = output_dir / ("stacked_images_" + speckle) / position
+
+    try:
+        stacked_image_path.mkdir(exist_ok=True, parents=True)
+    except OSError as error:
+        print (f"Creation of the directory {stacked_image_path} failed")
+        print(error)
+        sys.exit()
+    else:
+        return stacked_image_path
+
+
+def stack_trimmed_images(image_path_list, output_dir, speckle, position):
+    """Stacks trimmed images from provided image_path_list with rasterio in order to preserve the georeferencing
+    and writes both bands to tif files ("s1_stacked_VV.tif", "s1_stacked_VH.tif").
+
+    Parameters
+    ----------
+    image_path_list: List
+        List containing paths of images of the same position with different capture dates.
+    output_dir: Path
+        Path where the stacked image shall be stored
+    speckle: String
+        Set the desired speckle variant (options available in geocropper.stack_trimmed_images)
+    position: String
+        Set the position of the image in latitude and longitude format as a string (e.g. "-68.148781_44.77383")
+    """
+
+    rasterio_image_list = []
+    for image_path in image_path_list:
+        rasterio_image_list.append(rasterio.open(image_path))
+
+    try:
+        profile = rasterio_image_list[0].profile
+        profile.update({"count":len(rasterio_image_list)})
+
+        stacked_image_path = mkdir_for_stacked_image(output_dir, speckle, position)
+
+        for band, tif_file in enumerate(["s1_stacked_VV.tif", "s1_stacked_VH.tif"], start=1):
+            with rasterio.open((stacked_image_path / tif_file), "w", **profile) as dest:
+                for i, image in enumerate(rasterio_image_list, start=1):
+                    dest.write(image.read(band), i)
+    
+    except IndexError as error:
+        print (f"No images for stacking were found.")

```

Comment:
When would this occur?
Am I reading it correctly, that this error theoretically never should be thrown but it is kind of a safe net if there should be an unforeseen error?

Reply:
This error should occur if the provided source directory does not contain .tif files as expected by the code.